packages/@storefront/flux-capacitor/test/unit/core/emitter.ts

```diff
@@ -0,0 +1,188 @@
+
+      expect(emitter._barriers[key].cbs.length).to.equal(2);
+    });
+
+    it('should update the _lookups array with a new key', () => {
+      const events1 = ['a', 'b'];
+      const events2 = ['a', 'c'];
+      const key = 'a:b';
+      const key2 = 'a:c';
+
+      emitter.all(events1, () => ({}));
+      emitter.all(events2, () => ({}));
+
+      expect(emitter._lookups).to.eql({ a: [key, key2], b: [key], c: [key2] });
+    });
+
+    it('should allow the context to be specified' , () => {
+      const events1 = ['a', 'b'];
+      const key = 'a:b';
+      const cb = () => ({});
+      const that = { c: 'd' };
+
+      emitter.all(events1, cb, that);
+
+      expect(emitter._barriers[key]).to.eql({ events: { a: 0, b: 0 }, cbs: [{ cb, context: that }] });
+    });
+
+    it('should return the emitter' , () => {
+      expect(emitter.all(['a', 'b', 'c'], () => null)).to.eq(emitter);
+    });
+  });
+
+  describe('allOff', () => {
+    it('should remove the target callback', () => {
+      const events = ['a', 'b'];
+      const key = 'a:b';
+      const cb = () => {};
+      emitter._lookups = { a: [key], b: [key] };
+      emitter._barriers[key] = { cbs: [{ cb, context: emitter }] };
+
+      emitter.allOff(events, cb);
+
+      expect(emitter._barriers[key].cbs).to.eql([]);
+    });
+
+    it('it should not affect unrelated barriers', () => {
+      const events1 = ['a', 'b'];
+      const events2 = ['b', 'c'];
+      const key1 = 'a:b';
+      const key2 = 'b:c';
+      const cb = spy();
+      emitter.all(events1, cb);
+      emitter.all(events2, cb);
+
+      emitter.allOff(events1, cb);
+
+      expect(emitter._barriers[key2]).to.eql({
+        events: { b: 0, c: 0 },
+        cbs: [{ cb, context: emitter }],
+      });
+    });
+
+    it('should return the emitter', () => {
+      expect(emitter.allOff(['a', 'b', 'c'], () => null)).to.eq(emitter);
+    });
+  });
+
+  describe('emit', () => {
+    it('should update the _barriers counters if they exist', () => {
+      const events = ['a', 'b'];
+      const key = 'a:b';
+
+      emitter.all(events, () => {});
+      emitter.emit('a', null);
+
+      expect(emitter._barriers[key].events.a).to.equal(1);
+    });
+
+    it('should invoke the callbacks if each event in a given collection has been emitted at least once', () => {
+      const events = ['a', 'b'];
+      const cb1 = spy();
+      const cb2 = spy();
+
+      emitter.all(events, cb1);
+      emitter.all(events, cb2);
+      events.forEach(ev => emitter.emit(ev, null));
+
+      expect(cb1).to.have.been.called;
+      expect(cb2).to.have.been.called;
+    });
+
+    it('should reset the event counters for a given collection', () => {
+      const events = ['a', 'b'];
+      const key = 'a:b';
+      const cb = spy();
+
+      emitter.all(events, cb);
+      events.forEach(ev => emitter.emit(ev, null));
+
+      expect(Object.keys(emitter._barriers[key].events).map(k => emitter._barriers[key].events[k])).to.eql([0, 0]);
```

Comment:
Use `Object.values`. The order in which `Object.keys` and `Object.values` return their contents is unspecified, but in this case, it doesn't matter since their values should all be 0.

Reply:
As discussed, `Object.values` was introduce by the ES017 spec, which is not targeted by our `tsconfig` file. Leaving as-is.